### PR TITLE
refactor(w3c/informative): remove jQuery

### PR DESCRIPTION
**Pull request by saschanaz**
This one was easy.

### CODE REVIEW

File: src/w3c/informative.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 export const name = "w3c/informative";
 
-export function run(conf, doc, cb) {
-  $("section.informative")
-    .find("> h2:first, > h3:first, > h4:first, > h5:first, > h6:first")
-    .after("<p><em>This section is non-normative.</em></p>");
-  cb();
+export function run() {
+  Array.from(document.querySelectorAll("section.informative"))
+    .map(informative => Array.from(informative.children).find(child => child instanceof HTMLHeadingElement))
```
Comment (marcoscaceres) [inline src/w3c/informative.js]: Nice!

File: src/w3c/informative.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+    .filter(heading => heading)
+    .forEach(heading => {
+      heading.parentNode.insertBefore(hyperHTML`<p><em>This section is non-normative.</em></p>`, heading.nextSibling);
```
Comment (marcoscaceres) [inline src/w3c/informative.js]: Nit: this should be a reduce.

Comment (saschanaz) [reply src/w3c/informative.js]: I'm not very familiar with `.reduce`, how should the change look like?

File: src/core/base-runner.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ function toRunnable(plug) {
       }
       try {
         // Modern plugins are async or normal functions, take one argument (conf)
-        if (plug.run.length === 1) {
+        if (plug.run.length <= 1) {
```
Comment (marcoscaceres) [inline src/core/base-runner.js]: Nit, should update comment to say zero argument.

File: src/w3c/informative.js (hunk 2 above)
Comment (marcoscaceres) [reply src/w3c/informative.js]: Actually, you are correct here. I misread the code. No need for reduce.

File: src/w3c/informative.js (hunk 1 above)
Comment (marcoscaceres) [reply src/w3c/informative.js]: I’m now kinda unsure if we shouldn’t let a selector find the headings. In addition to the array creation, we are walking potentially all the children per section, which seems a little excessive. Wdyt?

Comment (saschanaz) [reply src/w3c/informative.js]: Hmm yes, that may be a problem, although I think an informative section should only contain ~10 direct descendants.

Comment (saschanaz) [reply src/w3c/informative.js]: My drive was that the original jQuery selector doesn't have a widely available standard equivalent.

```
> h2:first, > h3:first, > h4:first, > h5:first, > h6:first
```

We can just ignore `:first` part (which forces the selector to stop at the first match) but not `>`, which limits the selector for the direct descendants and has a non-widely available equivalent [`:scope`](https://developer.mozilla.org/en-US/docs/Web/CSS/:scope).

But I'm starting to think that we probably also can ignore `>` as anyway we shouldn't see an indirect child headings. Wdyt?

Comment (marcoscaceres) [reply src/w3c/informative.js]: All sections are supposed to have a heading (and ReSpec complains if they don't)... so, yeah, that could work. Let's try it.

File: src/core/base-runner.js [hunk 4]
```diff
@@ -27,8 +27,8 @@ function toRunnable(plug) {
         performance.mark(name + "-start");
       }
       try {
-        // Modern plugins are async or normal functions, take one argument (conf)
-        if (plug.run.length === 1) {
+        // Modern plugins are async or normal functions, take zero or one argument (conf)
+        if (plug.run.length <= 1) {
```
Comment (marcoscaceres) [inline src/core/base-runner.js]: nit: `<==` ... just as matter of style, we use strict equality throughout the project

**Comment by marcoscaceres** [comment]
Tiny nit aside, feel free to merge.

File: src/core/base-runner.js (hunk 4 above)
Comment (saschanaz) [reply src/core/base-runner.js]: I think `<==` is not even a JS operator...

Comment (marcoscaceres) [reply src/core/base-runner.js]: Argh... I quit.

Comment (saschanaz) [reply src/core/base-runner.js]: I haven't heard there is a strict comparison operator in JavaScript, we probably should just ignore the type strictness here.

Comment (marcoscaceres) [reply src/core/base-runner.js]: You are too kind @saschanaz. I just need more coffee.

**Pull request merged**